storage/source/StorageMetrics.cc

```diff
@@ -86,18 +86,62 @@ float StorageMetrics::blocksPerDataNodeSD() {
   return stDev(dataNodeBlockCounts);
 }
 
-float StorageMetrics::recoverySpeed() {
-  // TODO(ejd6): implement
-  return 0.0;
+int StorageMetrics::replicationRecoverySpeed() {
+  if (timeInProgress) {
+    LOG(ERROR) << "StorageMetrics: recovery timing already in progress";
+    return -1;
+  }
+  timeInProgress = true;
+  tempClock = clock();
+
+  int error;
+  std::vector<std::string> datanode_ids;
+  if (!zkWrapper->wget_children(
+      "/work_queues/replicate",
+      datanode_ids,
+      watcher_replicate,
+      this,
+      error)) {
+    LOG(ERROR) << "Storage metrics failed to set watcher.";
+    return -1;
+  }
+  LOG(INFO) << "STORAGE METRICS WATCHER set";
+  return 0;
+}
+
+void StorageMetrics::watcher_replicate(zhandle_t *zzh,
+                                       int type,
+                                       int state,
+                                       const char *path,
+                                       void *watcherCtx) {
+  LOG(INFO) << "STORAGE METRICS WATCHER TRIGGERED";
+  StorageMetrics *metrics = reinterpret_cast<StorageMetrics *>(watcherCtx);
+  std::shared_ptr<ZKWrapper> zkWrapper = metrics->zkWrapper;
+  int error;
+  std::vector<std::string> datanode_ids;
+  if (!zkWrapper->get_children("/work_queues/replicate", datanode_ids, error)) {
```

Comment:
This is just a simple thing, but instead of hard coding those paths, ZkClientCommon has those defined if you want to use ZkClientCommon::REPLICATE_QUEUES_NO_BACKSLASH.